IdentityCore/src/logger/MSIDLogger.m

```diff
@@ -156,74 +156,88 @@ - (void)logWithLevel:(MSIDLogLevel)level
+                NSAssert(NO, @"Correlation ID not of NSUUID class");
+                correlationIdStr = @"[Invalid non-NSUUID correlationID]";
             }
+        }
+        else if (context)
+        {
+            correlationIdStr = [NSString stringWithFormat:@" - %@", [context correlationId]];
+        }
+        
+        NSString *dateStr = [s_dateFormatter stringFromDate:[NSDate date]];
+        
+        NSString *sdkName = [MSIDVersion sdkName];
+        NSString *sdkVersion = [MSIDVersion sdkVersion];
+        
+        NSString *sourceInfo = @"";
+        if (self.sourceLineLoggingEnabled && filename.length)
+        {
+            sourceInfo = [NSString stringWithFormat:@" %@:%lu: %@", filename.lastPathComponent, (unsigned long)lineNumber, function];
+        }
+        
+        __auto_type threadName = [[NSThread currentThread] isMainThread] ? @" (main thread)" : nil;
+        if (!threadName) {
+            threadName = [NSThread currentThread].name ?: @"";
+        }
+        
+        __auto_type threadInfo = [[NSString alloc] initWithFormat:@"TID=%llu%@", tid, threadName];
+        
+        if (self.nsLoggingEnabled)
+        {
+            NSString *logLevelStr = [self stringForLogLevel:self.level];
             
-            __auto_type threadName = [[NSThread currentThread] isMainThread] ? @" (main thread)" : nil;
-            if (!threadName) {
-                threadName = [NSThread currentThread].name ?: @"";
-            }
+            NSString *log = [NSString stringWithFormat:@"%@ %@ %@ %@ [%@%@]%@ %@:%@ %@", threadInfo, sdkName, sdkVersion, [MSIDDeviceId deviceOSId], dateStr, correlationIdStr, componentStr, logLevelStr, sourceInfo, message];
             
-            __auto_type threadInfo = [[NSString alloc] initWithFormat:@"TID=%llu%@", tid, threadName];
+            NSLog(@"%@", log);
+        }
+        
+        if (self.callback || self.loggerConnector)
+        {
+            NSString *log = [NSString stringWithFormat:@"%@ %@ %@ %@ [%@%@]%@%@ %@", threadInfo, sdkName, sdkVersion, [MSIDDeviceId deviceOSId], dateStr, correlationIdStr, componentStr, sourceInfo, message];
             
-            if (self.nsLoggingEnabled)
-            {
-                NSString *logLevelStr = [self stringForLogLevel:self.level];
-                
-                NSString *log = [NSString stringWithFormat:@"%@ %@ %@ %@ [%@%@]%@ %@:%@ %@", threadInfo, sdkName, sdkVersion, [MSIDDeviceId deviceOSId], dateStr, correlationIdStr, componentStr, logLevelStr, sourceInfo, message];
-                
-                NSLog(@"%@", log);
-            }
+            BOOL lineContainsPII = self.piiLoggingEnabled ? containsPII : NO;
             
-            if (self.callback || self.loggerConnector)
+            if (self.callback) self.callback(level, log, lineContainsPII);
+            if (self.loggerConnector) [self.loggerConnector onLogWithLevel:level lineNumber:lineNumber function:function message:log];
```

Comment:
should this be an "else condition" or is it possible to have both callback and loggerConnector?